### ISSUE TITLE
xLights: State Definition Window does not update properly when adding a new definition

### ISSUE THREAD

**Issue by Holiday851** [smeighan/xLights#3155]
2022.06 on Windows 10 64-bit

Repro that makes it more obvious what's going on:
1. Create a new model
2. Edit the States
3. Click the "Add" button and create a new State Definition called "A." Notice that the combo box says "Node Ranges" and the "Color" column is showing. I believe that both of these are incorrect.
4. Click in the first column of the first row and name the state "a" (NOTE: is there a reason you cannot use capital letters here?) 
5. Click the "Add" button and create a new State Definition called "B"
6. Notice that the list box still says "a"
![StateDefinition1](https://user-images.githubusercontent.com/95460274/156959986-da9fe077-6551-4ac2-b530-35718ca6d647.png)
7. From the "Name" combo box, select "A". Notice it now correctly says "Single Nodes" and has no "Color" column. The listbox has the "a" entry.
8. From the "Name" combo box, select "B." Notice that now the "a" in the listbox goes away and it has the correct state of "B".

Expected:
When a new State Definition is created, the proper state of the new definition shows up in the data area (i.e., Single Nodes, no items in the listbox, and no Color column) instead of leaving whatever was there before.

Note:
Data entry gets a little weird since the actual state of the new definition doesn't match the UI. For example, after creating "B," (right after step 6) if you then double-click the second column to bring up the Node Selection and select some nodes, then switch back to "A" and back to "B" again, you will get this, which does not match, but I believe that that will be okay once the data area is updated to the full state of the newly created item.
![StateDefinition2](https://user-images.githubusercontent.com/95460274/156960294-b50c609b-2118-4dd3-98b3-f701a6aea581.png)

Video showing the above repro. If you can't see it from within the bug, download and try. I can't see it from within GitHub (it plays - just there's nothing to see), and I don't know if it behaves that way for everyone.

https://user-images.githubusercontent.com/95460274/156960682-c564359e-7d46-4ca3-a8e0-e1d8041c3f9e.mp4

----------------
Leaving the original text of the bug I logged below, but I think it's just a symptom of the above.

This is completely cosmetic (i.e., super low priority), but I noticed it when I was trying to repro a crash in the State Definition window.

Repro:
1. Create a new model 
2. Edit the States.
3. Add a new State.
4. Notice that the "Force Custom Colors" checkbox is not checked but the "Color" column is present.
5. Check the checkbox. 
6. Uncheck the checkbox. Notice that the Color column goes away.

Note: 
If the model already contains a state definition when the "State Definition" window is opened, it does NOT repro.

Expected:
The state of the "Force Custom Colors" checkbox to match the visibility of the "Color" column

![ColorColumnShowsWhenNotColorForced](https://user-images.githubusercontent.com/95460274/156955358-283f6d09-e8c8-4b2b-9870-838135ccd65e.png)

https://user-images.githubusercontent.com/95460274/156955154-be517500-19fc-4ef6-a03d-d2a8966410a0.mp4

May need to download the video in order to see it (I cannot see it from within the bug, but it works fine when I play it locally)
